### PR TITLE
Mongo client to string

### PR DESCRIPTION
**Pull request by DerThorsten**
this change was needed to build this (on apple silicon) with clang when using MongoDB from conda forge.
Following https://www.mongodb.com/community/forums/t/no-implicit-conversion-from-b-string-to-string-on-arrays/215020

### CODE REVIEW

File: cpp/vcpkg
Comment (qc00) [inline cpp/vcpkg:1]: Accidentally committed?

Comment (DerThorsten) [reply cpp/vcpkg:1]: oh jeah, sorry

File: cpp/arcticdb/storage/mongo/mongo_client.cpp [hunk 1]
```diff
@@ -29,7 +30,7 @@ namespace detail {
 template<typename ElementType>
 std::string get_string_element(const ElementType& element) {
 #if (MONGOCXX_VERSION_MAJOR * 1000 + MONGOCXX_VERSION_MINOR) >= 3007
-    return std::string(element.get_string());
+    return bsoncxx::string::to_string(element.get_string().value);
```
Comment (qc00) [inline cpp/arcticdb/storage/mongo/mongo_client.cpp:33]: I still wonder why this is necessary. BSONCXX's `b_string` has an [`operator stdx::string_view() const`](https://github.com/mongodb/mongo-cxx-driver/blame/master/src/bsoncxx/types.hpp#LL141C20-L141C54).

In the environments where `stdx == std`, [`std::string` ctor (10)](https://en.cppreference.com/w/cpp/string/basic_string/basic_string) should apply. If `stdx != std`, then `bsoncxx::string::to_string` will not return `std::string` so an implicit conversion is being done on return.

**Comment by DerThorsten** [comment]
@qc00 fyi this is the error I see when I use `return std::string(element.get_string());`
```
ArcticDB/cpp/arcticdb/storage/mongo/mongo_client.cpp:33:12: error: no matching conversion for functional-style cast from 'types::b_string' to 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>')
    return std::string(element.get_string());
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ArcticDB/cpp/arcticdb/storage/mongo/mongo_client.cpp:58:23: note: in instantiation of function template specialization 'arcticdb::storage::mongo::detail::get_string_element<bsoncxx::document::element>' requested here
        start_index = get_string_element(doc["start_key"]);
                      ^
ArcticDB/cpp/arcticdb/storage/mongo/mongo_client.cpp:415:32: note: in instantiation of function template specialization 'arcticdb::storage::mongo::detail::atom_key_from_document<const bsoncxx::document::view>' requested here
            auto key = detail::atom_key_from_document(doc, key_type);
                               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:816:66: note: candidate constructor not viable: no known conversion from 'types::b_string' to 'const std::basic_string<char>::allocator_type' (aka 'const std::allocator<char>') for 1st argument
    _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_AFTER_CXX17 explicit basic_string(const allocator_type& __a)
                                                                 ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:823:35: note: candidate constructor not viable: no known conversion from 'types::b_string' to 'const std::string' for 1st argument
    _LIBCPP_CONSTEXPR_AFTER_CXX17 basic_string(const basic_string& __str);
                                  ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:828:5: note: candidate constructor not viable: no known conversion from 'types::b_string' to 'std::string' for 1st argument
    basic_string(basic_string&& __str)
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:841:5: note: candidate constructor template not viable: no known conversion from 'types::b_string' to 'const char *' for 1st argument
    basic_string(const _CharT* __s) : __r_(__default_init_tag(), __default_init_tag()) {
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:895:5: note: candidate constructor not viable: no known conversion from 'types::b_string' to 'initializer_list<char>' for 1st argument
    basic_string(initializer_list<_CharT> __il);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:881:18: note: candidate template ignored: requirement '__can_be_converted_to_string_view<char, std::char_traits<char>, bsoncxx::types::b_string>::value' was not satisfied [with _Tp = bsoncxx::types::b_string]
        explicit basic_string(const _Tp& __t);
                 ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:813:57: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
    _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_AFTER_CXX17 basic_string()
                                                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:824:35: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    _LIBCPP_CONSTEXPR_AFTER_CXX17 basic_string(const basic_string& __str, const allocator_type& __a);
                                  ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:836:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    basic_string(basic_string&& __str, const allocator_type& __a);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:849:9: note: candidate constructor template not viable: requires 2 arguments, but 1 was provided
        basic_string(const _CharT* __s, const _Allocator& __a);
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:856:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    basic_string(const _CharT* __s, size_type __n);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:860:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    basic_string(size_type __n, _CharT __c);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:885:18: note: candidate constructor template not viable: requires 2 arguments, but 1 was provided
        explicit basic_string(const _Tp& __t, const allocator_type& __a);
                 ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:889:9: note: candidate constructor template not viable: requires 2 arguments, but 1 was provided
        basic_string(_InputIterator __first, _InputIterator __last);
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:897:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    basic_string(initializer_list<_CharT> __il, const _Allocator& __a);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:791:14: note: candidate constructor not viable: requires 3 arguments, but 1 was provided
    explicit basic_string(__uninitialized_size_tag, size_type __size, const allocator_type& __a)
             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:858:5: note: candidate constructor not viable: requires 3 arguments, but 1 was provided
    basic_string(const _CharT* __s, size_type __n, const _Allocator& __a);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:864:9: note: candidate constructor template not viable: requires 3 arguments, but 1 was provided
        basic_string(size_type __n, _CharT __c, const _Allocator& __a);
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:870:5: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
    basic_string(const basic_string& __str, size_type __pos,
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:892:9: note: candidate constructor template not viable: requires 3 arguments, but 1 was provided
        basic_string(_InputIterator __first, _InputIterator __last, const allocator_type& __a);
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:867:5: note: candidate constructor not viable: requires at least 3 arguments, but 1 was provided
    basic_string(const basic_string& __str, size_type __pos, size_type __n,
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/string:875:9: note: candidate constructor template not viable: requires at least 3 arguments, but 1 was provided
        basic_string(const _Tp& __t, size_type __pos, size_type __n,
        ^
ArcticDB/cpp/arcticdb/storage/mongo/mongo_client.cpp:43:21: error: no matching function for call to 'get_string_element'
        stream_id = get_string_element(doc["stream_id"]);
                    ^~~~~~~~~~~~~~~~~~
ArcticDB/cpp/arcticdb/storage/mongo/mongo_client.cpp:62:22: note: in instantiation of function template specialization 'arcticdb::storage::mongo::detail::stream_id_from_document<const bsoncxx::document::view>' requested here
    auto stream_id = stream_id_from_document(doc, key_type);
                     ^
ArcticDB/cpp/arcticdb/storage/mongo/mongo_client.cpp:415:32: note: in instantiation of function template specialization 'arcticdb::storage::mongo::detail::atom_key_from_document<const bsoncxx::document::view>' requested here
            auto key = detail::atom_key_from_document(doc, key_type);
                               ^
ArcticDB/cpp/arcticdb/storage/mongo/mongo_client.cpp:31:13: note: candidate template ignored: substitution failure [with ElementType = bsoncxx::document::element]
std::string get_string_element(const ElementType& element) {
            ^
2 warnings and 2 errors generated.
make[2]: *** [arcticdb/CMakeFiles/arcticdb_core_object.dir/storage/mongo/mongo_client.cpp.o] Error 1
make[1]: *** [arcticdb/CMakeFiles/arcticdb_core_object.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by qc00** [comment]
I guess Conda's build of `mongocxx` does not use `std::*` as the implementation of [`stdx::string_view`](https://github.com/mongodb/mongo-cxx-driver/blob/master/src/bsoncxx/stdx/string_view.hpp), so the direct conversion is not working...

**Review by qc00 — APPROVED** [review]

**Review by poodlewars — APPROVED** [review]

**Comment by DerThorsten** [comment]
using #302

**Pull request closed without merging**